### ISSUE TITLE
invalid-name check for TypeAlias should allow for digits in names

### ISSUE THREAD

**Issue by sodul** [PyCQA/pylint#8485]
### Bug description

from typing import Union

Ec2Data = Union[int, str]

### Command used

```shell
pylint ec2.py
```

### Pylint output

```shell
Type alias name "Ec2Data" doesn't conform to predefined naming style (3:0) [invalid-name]
```

### Expected behavior

Digits in variable names are pretty common and should be allowed. In the example above we work with EC2 and there is not good substitute for using a digit.

We believe pylint is being too strict in the new check introduced in #7116.

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.11.2 (main, Mar 20 2023, 21:17:15) [Clang 14.0.0 (clang-1400.0.29.202)]
```

### OS / Environment

MacOS 13.2.1

**Comment by DanielNoord** [comment]
You are correct! Would you want to open a PR for this? You have identified the correct PR. We can just change the regex and add a new test case.

**Comment by sodul** [comment]
@DanielNoord I noticed that the pattern is inconsistent with the other invalid-name rules. For example we can have `AWSClient` classes but `AWSClientAlias` gets rejected as a type alias. I'm personally in favor of renaming `AWSClient` to `AwsClient` but the fact that the rules are inconsistent is not ideal IMHO.

**Comment by sodul** [comment]
While looking at the code I noticed that TypeVars have the same issue with integers. It seems a tad bit more complicated to fix though. I can open a separate bug report for it but I won't have time to send a PR in the short term.

**Comment by DanielNoord** [comment]
> While looking at the code I noticed that TypeVars have the same issue with integers. It seems a tad bit more complicated to fix though. I can open a separate bug report for it but I won't have time to send a PR in the short term.

Let's create another issue to track and fix this!
